internal/charmstore/elasticsearch.go

```diff
@@ -68,6 +68,11 @@ const esMappingJSON = `
           }
         }
       },
+      "PromulgatedURL" : {
+        "type" : "string",
+        "index": "not_analyzed",
+        "index_options" : "docs"
+      },
```

Comment:
The URL supports ngrams. It's probably ok to not include ngrams in PromulgatedURL as we'll find the name by the URL and boost up due to promulgation. Just a thought on the schema.